modules/openapi-generator/src/main/resources/dart2/class.mustache

```diff
@@ -1,10 +1,14 @@
 class {{classname}} {
   {{#vars}}
-  {{#description}}/* {{{description}}} */{{/description}}
-  {{{dataType}}} {{name}} = {{{defaultValue}}};
+  {{#description}}/// {{{description}}}{{/description}}
+  {{^isEnum}}
+  {{^defaultValue}}{{{dataType}}} {{name}};{{/defaultValue}}{{#defaultValue}}{{{dataType}}} {{name}} = {{defaultValue}};{{/defaultValue}}
+  {{/isEnum}}
+  {{#isEnum}}
   {{#allowableValues}}
-  {{#min}} // range from {{min}} to {{max}}{{/min}}//{{^min}}enum {{name}}Enum { {{#values}} {{.}}, {{/values}} };{{/min}}{
+  {{#min}} // range from {{min}} to {{max}}{{/min}}{{classname}}{{datatypeWithEnum}} {{name}}{{#required}} = {{{defaultValue}}}{{/required}}{{^required}}{{/required}};
```

Comment:
@agilob Please use `{{{datatypeWithEnum}}}` instead of `{{datatypeWithEnum}}` as now I got the following:
```
  IntersectionList&lt;EntryEnum&gt; entry;
```

Reply:
I fixed it per your comment. I dont know how to reproduce your results, CircleCI also produces same code as was on my side, unreliable test/environment?

Reply:
dart2 uses the petstore.yaml for tests but pestore.yaml only covers the very basic test cases. [Fake Petstore](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/petstore-with-fake-endpoints-models-for-testing.yaml) contains a lot more test cases, which should cover this particular issue.

For my findings above, I was using https://github.com/geoDavey/osrm-openapi/blob/master/osrm-openapi.yaml suggested by @sbu-WBT

Reply:
Updated using your file @sbu-WBT